conjureup/utils.py

```diff
@@ -136,19 +135,18 @@ def last_ten_lines(s):
             async with aiofiles.open(step_path + '.out', 'r') as f:
                 while proc.returncode is None:
                     async for line in f:
-                        if line.startswith('result:'):
-                            result = line
-                        msg = line
                         msg_cb(line)
                     await asyncio.sleep(0.01)
 
             if proc.returncode != 0:
-                raise Exception("Failure in step {}".format(step_file))
+                raise Exception("Failure in step {}".format(step.path))
 
             if event_name is not None:
                 track_event(event_name, "Done", "")
 
-    return result if result else msg
+    result = app.state.get("conjure-up.{}.{}".format(app.config['spell'],
+                                                     step.result_key))
```

Comment:
Probably support not having a message at all, less to confuse the user with. What do you think?